src/utils/input-processor.ts

```diff
@@ -0,0 +1,39 @@
+import store from "@/store";
+
+const timestampFromDate = (date: string): number => {
+  return new Date(date).getTime() / 1000;
+};
+
+const getNetworkTimestamp = (date: string): number => {
+  const epochTimestamp = timestampFromDate(store.getters["network/epoch"]);
+
+  return timestampFromDate(date) - epochTimestamp;
+};
+
+const lookupPublicKey = (username: string): string | null => {
+  const lookup = store.getters["delegates/byUsername"](username);
+
+  return lookup ? lookup.publicKey : null;
+};
+
+export const inputProcessor = (inputName: string, inputValue: any): string | number => {
+  let value = inputValue;
+
+  const arktoshiValues = ["amount", "totalAmount", "fee", "totalFee", "reward", "balance"];
+  const publicKeyValues = ["vote", "generatorPublicKey"];
+
+  if (arktoshiValues.find(name => inputName.includes(name))) {
```

Comment:
why an `includes` instead of an `===`? what are the names of the inputs? 🤔

Reply:
also if `inputName` would match one of the options, it would be possible to use `arktoshiValues.includes(inputName)` instead

Reply:
The reason for this is because the `inputName` that is passed to the function does not always exactly match the entries in that arktoshi/publickey value arrays. This is the case for fields that have a 'from' and 'to' child. For example, when processing the 'amount from' value, the inputname would be `amount-from`.

The component that utilizes this input processor could potentially be refactored to accomodate for this suggestion, if needed.

Reply:
I see, you can leave it as is in that case 🤔